Ginger/Ginger/Run/RunSetsExecutionsHistoryPage.xaml.cs

```diff
@@ -333,40 +416,296 @@ private async void LoadExecutionsHistoryData()
                 }
             }
 
-            grdExecutionsHistory.DataSourceList = executionsHistoryListSortedByDate;
-            grdExecutionsHistory.Visibility = Visibility.Visible;
+            xGridExecutionsHistory.DataSourceList = executionsHistoryListSortedByDate;
+            xGridExecutionsHistory.Visibility = Visibility.Visible;
             Loading.Visibility = Visibility.Collapsed;
         }
+        /// <summary>
+        /// Loads the execution history data using GraphQL.
+        /// </summary>
+        private async Task LoadExecutionsHistoryDataGraphQl()
+        {
+            xGridExecutionsHistory.Visibility = Visibility.Collapsed;
+            Loading.Visibility = Visibility.Visible;
+            mExecutionsHistoryList.Clear();
+            await LoadRemoteDataFromGraphQL();
+            SetContentInGrid();
+
 
-        private void AddRemoteExucationRunsetData()
+        }
+        /// <summary>
+        /// Sets the content in the grid by populating the data source list with sorted and filtered execution history.
+        /// </summary>
+        void SetContentInGrid()
         {
-            List<RunSetReport> runsetsReport = new List<RunSetReport>();
-            if (RunsetConfig != null)
+            ObservableList<RunSetReport> executionsHistoryListSortedByDate = new ObservableList<RunSetReport>();
+            if (mExecutionsHistoryList != null && mExecutionsHistoryList.Count > 0)
             {
-                runsetsReport = new GingerRemoteExecutionUtils().GetRunsetExecutionInfo(WorkSpace.Instance.Solution.Guid, RunsetConfig.Guid);
+                IEnumerable<RunSetReport> sortedAndFilteredExecutionHistoryList = mExecutionsHistoryList
+                   .Where(report => report.RunSetExecutionStatus != eRunStatus.Automated)
+                   .OrderByDescending(item => item.StartTimeStamp);
+
+                foreach (RunSetReport runSetReport in sortedAndFilteredExecutionHistoryList)
+                {
+                    runSetReport.StartTimeStamp = runSetReport.StartTimeStamp.ToLocalTime();
+                    runSetReport.EndTimeStamp = runSetReport.EndTimeStamp.ToLocalTime();
+                    executionsHistoryListSortedByDate.Add(runSetReport);
+                }
             }
-            else
+
+            xGridExecutionsHistory.DataSourceList = executionsHistoryListSortedByDate;
+            xGridExecutionsHistory.Visibility = Visibility.Visible;
+            Loading.Visibility = Visibility.Collapsed;
+        }
+        /// <summary>
+        /// Loads remote data from GraphQL and updates the execution history page.
+        /// </summary>
+        private async Task LoadRemoteDataFromGraphQL()
+        {
+            try
+            {
+                int recordLimit = (int)xPageSizeComboBox.SelectedItem;
+                graphQlClient.ResetPagination();
+
+                response = RunsetConfig == null
```

Comment:
Use if (mExecutionHistoryLevel == eExecutionHistoryLevel.Solution) to check level of execution history page

Reply:
done